### ISSUE TITLE
[Bug]: Recommended way of packaging stream schemas as JSON files should use `importlib.resources`

### ISSUE THREAD

**Issue by edgarrmondragon** [meltano/sdk#875]
### Singer SDK Version

latest

### Python Version

NA

### Bug scope

Taps (catalog, state, stream maps, etc.)

### Operating System

Any

### Description

The currently recommended way (as used in the cookiecutter templates) to ship JSON files for the stream schemas relies on `__file__`, which doesn't play well with modern tools for packaging a Python application (such as a Singer tap/target) as a standalone binary, such as [PyOxidizer](https://pyoxidizer.readthedocs.io/en/stable/).

The solution is to use `importlib.resources` or the backport package `importlib_resources` on Python < 3.9.

This would also require changing the following:

1. drop support for raw string file path
2. discourage use of `Path` instances
3. change our type hints to use `importlib.resources.abc.Traversable`
4. add an `__init__.py` to the `schemas/` directory shipped with the cookiecutter

For a example, see https://github.com/edgarrmondragon/tap-bitso/pull/125. The PR is currently failing static type checks because type annotations are not compatible with the latest SDK.

### Code

```diff
+ from my_tap import schemas  # assumes a __init__.py exists in my_tap/schemas/
+ 
+ if sys.version_info >= (3, 9):
+     import importlib.resources as resources
+ else:
+     import importlib_resources as resources

- SCHEMAS_DIR = Path(__file__).parent / "./schemas"
+ SCHEMAS_DIR = resources.files(schemas)
```

**Comment by edgarrmondragon** [comment]
Still relevant

**Comment by edgarrmondragon** [comment]
Closed by #2130
